### ISSUE TITLE
Speed test

### ISSUE THREAD

**Issue by vindolineKis** [vindolineKis/QCircLearning#23]

**Comment by vindolineKis** [comment]
1. Each epoch takes 0.02s–0.09s; on average, 500 epochs require 10–45 seconds. A single backward search takes 0.01s–0.06s. For a training example with 100 iterations, the total training time is about 30 minutes to 1 hour.

> Possible methods for speed up：
> Reduce the number of epochs and lower the patience (accuracy threshold for neural network fitting).
> Increase the number of real data points added in each iteration (reducing the number of evaluations).


2. The utilization rate of server cores greatly impacts the training time of the neural network.
